experimental/packages/proto/src/common/internal.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { HrTime, SpanAttributes } from '@opentelemetry/api';
+import { fromByteArray } from 'base64-js';
+import { IAnyValue, IKeyValue } from './types';
+
+export function toAttributes(
+  attributes: SpanAttributes
+): IKeyValue[] {
+  return Object.keys(attributes).map(key => toKeyValue(key, attributes[key]));
+}
+
+export function toKeyValue(
+  key: string,
+  value: unknown
+): IKeyValue {
+  return {
+    key: key,
+    value: toAnyValue(value),
+  };
+}
+
+export function toAnyValue(value: unknown): IAnyValue {
+  if (typeof value === 'string') return { stringValue: value };
+  if (typeof value === 'number' && !Number.isInteger(value)) return { doubleValue: value };
+  if (typeof value === 'number' && Number.isInteger(value)) return { intValue: value };
+  if (typeof value === 'boolean') return { boolValue: value };
+  if (value instanceof Uint8Array) return { bytesValue: fromByteArray(value) };
```

Comment:
If we can distinguish between converting to JSON or protobuf like said in the OP, seems like we don't need to encode bytes values with base64 on targeting protobuf as protobufjs is going to decode it anyway.

Reply:
So you are suggesting to leave it as a Uint8Array and in the JSON exporter do the base64 conversion?

Reply:
Yes, that would be most optimal if possible.

Reply:
329678cf603677b26119a317ebd899365944a406